wolfcrypt/src/port/iotsafe/iotsafe.c

```diff
@@ -488,11 +493,28 @@ static int iotsafe_readfile(uint8_t *file_id, uint16_t file_id_sz,
                 return -1;
             }
             off += ret/2;
+#ifdef IOTSAFE_NO_GETDATA
+            if (XSTRNCMP(&resp[ret-4], "0000", 4) == 0) {
+                /* Strip trailing zeros */
+                for (int idx = 0; idx < off-1; idx+=2) {
```

Comment:
Do not declare the `int` in the for loop. Please move to top of brace section. Some compilers do not like this.